classes/class-wc-connect-stripe.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+if ( ! class_exists( 'WC_Connect_Stripe' ) ) {
+
+	class WC_Connect_Stripe {
+
+		/**
+		 * @var WC_Connect_API_Client
+		 */
+		private $api;
+
+		/**
+		 * @var WC_Connect_Options
+		 */
+		private $options;
+
+		/**
+		 * @var WC_Connect_Logger
+		 */
+		private $logger;
+
+		public function __construct( WC_Connect_API_Client $client, WC_Connect_Options $options, WC_Connect_Logger $logger ) {
+			$this->api = $client;
+			$this->options = $options;
+			$this->logger = $logger;
+		}
+
+		public function get_settings( $return_url ) {
+			$email = get_option( 'admin_email' );
+			$country = WC()->countries->get_base_country();
+			$result = $this->api->get_stripe_oauth_init( $return_url );
+
+			if ( is_wp_error( $result ) ) {
+				return $result;
+			}
+
+			set_transient( 'wcs_stripe_state', $result->state );
+
+			return array (
+				'email' => $email,
+				'country' => $country,
+				'oauthUrl' => $result->oauthUrl,
+			);
+		}
+
+		public function create_account( $email, $country ) {
+			$response = $this->api->create_stripe_account( $email, $country );
+			if ( is_wp_error( $response ) ) {
+				return $response;
+			}
+			return $this->save_stripe_keys( $response );
+		}
+
+		public function connect_oauth( $state, $code ) {
+			if ( $state !== get_transient( 'wcs_stripe_state' ) ) {
```

Comment:
Can you explain the reasoning behind this `state` verification? Does storing the state locally avoid having to implement a verification endpoint in our service layer?

If this should remain on the client, let's use the options API instead of transients since transients are never guaranteed to persist.

Reply:
We definitely want to store this state value temporarily. A transient isn't the right place for this since the state should be tied to the user. I'm thinking this is better placed in a session variable.